pkg/plugins/resources/yaml/main.go

```diff
@@ -52,28 +51,50 @@ func New(spec interface{}) (*Yaml, error) {
 		contentRetriever: &text.Text{},
 	}
 
-	newResource.spec.File = strings.TrimPrefix(newResource.spec.File, "file://")
-
-	err = newResource.Validate()
+	err = newResource.spec.Validate()
 	if err != nil {
 		return nil, err
 	}
 
+	newResource.files = make(map[string]string)
+	// File as unique element of newResource.files
+	if len(newResource.spec.File) > 0 {
+		newResource.files[strings.TrimPrefix(newResource.spec.File, "file://")] = ""
+	}
+	// Files
+	for _, filePath := range newResource.spec.Files {
+		newResource.files[strings.TrimPrefix(filePath, "file://")] = ""
+	}
+
 	return newResource, nil
+}
+
+func hasDuplicates(values []string) bool {
+	uniqueValues := make(map[string]string)
+	for _, v := range values {
+		uniqueValues[v] = ""
+	}
 
+	return len(values) != len(uniqueValues)
 }
 
 // Validate validates the object and returns an error (with all the failed validation messages) if it is not valid
-func (y *Yaml) Validate() error {
+func (s *Spec) Validate() error {
 	var validationErrors []string
 
 	// Check for all validation
-	if y.spec.File == "" {
-		validationErrors = append(validationErrors, "Invalid spec for yaml resource: 'file' is empty.")
+	if len(s.Files) == 0 && len(s.File) == 0 {
```

Comment:
```suggestion
	if len(s.Files) == 0 && s.File == "" {
```

The compiled code [is the same](https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go/54804858#54804858) but in this context, it's clearer to mark that `s.File` is a string (using length is not explicit: it could be a collection)

Reply:
Is there something you expect me to do here or is it just a comment on the change?

Reply:
no it is a code suggestion to change all the `len(s.File) == 0` occurences that this PR added/updated to `s.File == ""`